packages/story-editor/src/components/canvas/mediaRecordingLayer.js

```diff
@@ -168,80 +101,47 @@ function MediaRecordingLayer() {
     }
   }, [isReady, updateMediaDevices]);
 
-  useEffect(() => {
-    // Checking for srcObject avoids flickering due to the stream changing constantly.
-    if (streamRef.current && !streamRef.current.srcObject && liveStream) {
-      streamRef.current.srcObject = liveStream;
-    }
-
-    if (streamRef.current && !liveStream) {
-      streamRef.current.srcObject = null;
-    }
-  }, [liveStream]);
-
-  if (needsPermissions) {
-    return <PermissionsDialog />;
-  }
-
-  if (isFailed) {
-    return <ErrorDialog />;
-  }
+  // Video data was designed for a different purpose, so we need to fake the api a bit here
+  const videoData = useMemo(() => {
+    return isTrimming
+      ? {
+          element: {
+            width: FULLBLEED_RATIO * 480,
+            height: 480,
+            scale: 1,
+            focalX: 50,
+            focalY: 50,
+            flip: {},
+          },
+          resource: {
+            src: mediaBlobUrl,
+            length: duration,
```

Comment:
Length formated?

Reply:
This is only to fool the `generateVideoStrip` function into thinking, that it's a "real" resource. [That function only extracts these two values](https://github.com/GoogleForCreators/web-stories-wp/blob/e00e50da7a02d956b993f4038ca1c15d46eda3ef/packages/media/src/generateVideoStrip.js#L43).